### PR TITLE
Add file listing example

### PR DESCRIPTION
**Pull request by Sija**
New example listing recursively all files on a first found `Camera`

### CODE REVIEW

**Comment by zaeleus** [comment]
Can you make the output a flat list instead of a tree? For example,

```
/ (0 files)

/store_00010001 (0 files)

/store_00010001/DCIM (0 files)

/store_00010001/DCIM/101APPLE (409 files)
IMG_1001.JPG
IMG_1002.MOV
IMG_1003.JPG
IMG_1004.JPG
...

/store_00010001/DCIM/100APPLE (365 files)
IMG_0004.JPG
IMG_0014.JPG
IMG_0015.JPG
IMG_0017.JPG
...
```

**Comment by Sija** [comment]
@zaeleus done!

File: examples/list_files.rb [hunk 1]
```diff
@@ -1,17 +1,15 @@
...
+# List all files
 
+def visit(folder)
   files = folder.files
-  puts "#{indent}#{folder.root? ? "/ (root)" : folder.path} (#{files.size} files)"
+  puts "#{folder.root? ? "/ (root)" : folder.path} (#{files.size} files)"
 
-  folder.folders.each { |child| visit(child, level + 1) }
-  indent << '  '
+  folder.folders.each { |child| visit(child) }
```
Comment (zaeleus) [inline examples/list_files.rb]: This traversal should be after completely visiting the node, else you end up printing a series of empty lines coming up from the stack.

**Comment by zaeleus** [comment]
Thanks!

**Pull request merged**